### DISCUSSION TITLE
How do I use the plugin to retrieve a payment token only?

### DISCUSSION THREAD

**Discussion by pma07pg** [google-pay/flutter-plugin#148] [Q&A]
If my app user wishes to have google pay as their preferred payment option, then I'd need to generate a token (not take payment) and use this with my payment merchant to generate the merchant specific token for future payments. Whenever the user causes a payment to be processed, then I can use this token in my backend to take payment. 

I see that the token is returned in the payment response, could I just use this payment token and can I send a payment amount of 0.00?

Thanks!

**Comment by cookii-king** [comment]
I used the nonce token to create a payment method with Braintree and it worked fine.
